Force Always on Top to be persisted

mantlabs opened the pull request:
## Description

Regarding this issue #6231, I agree that the nature of the "Always on Top" option is to be persisted like in KeePass.
However, in order to keep the current functionality intact, and let the user decide how it will work, I made a pull request with an additional option in:
Settings -> General (Basic Settings) -> User Interface: "Force Always on Top to be persisted". Note that this option only allows persistance of the existing "Always on Top" in View submenu.

This gives the ability to the user to persist the "Always on Top" setting on "View" menu. 

## Screenshots
![screenshot_persist_always_on_top](https://user-images.githubusercontent.com/76261135/110214865-511a8080-7eaf-11eb-942b-d37d414a4d6c.jpg)

## Notable code changes
Additional Settings in .ini
- "Always on Top" -> "GUI_AlwaysOnTop"
- "Force Always on Top to be persisted" -> "GUI_PersistAlwaysOnTop"

## Testing strategy
- Open the application with both settings false (default). Close the application. Re-open it. It works as in the initial state and works with "Always on Top" disabled.
- Open the application with only "Always on Top" setting checked. Close the application. Re-open it. The "Always on Top" is reset to false and works with it disabled.
- Open the application with only "Always on Top" setting checked. Check also the "Force Always on Top to be persisted". Close the application. Re-open it. The "Always on Top" is fetched from saved settings and is checked. "Always on Top" is enabled.
- Open the application with "Always on Top" setting un-checked. Check the "Force Always on Top to be persisted". Close the application. Re-open it. The "Always on Top" remains disabled and un-checked.

- Built successfully via cmake based on "develop" branch.
- Run tests (all passed) with:
make test ARGS+="--output-on-failure"

## Type of change
- ✅ New feature (change that adds functionality)

droidmonkey commented:
I'm not adding yet another option for this. Just save the always on top setting.

phoerious commented:
I am against adding this, but I am definitely against adding it as another option.

mantlabs commented:
> 
> 
> I'm not adding yet another option for this. Just save the always on top setting.

I thought you didn't want to change the existing functionality, that's why I introduced a new option. I reverted the new option and made the existing option to be saved as you told me.

If you would like to add it or not, it's up to you. In my opinion this is the right approach, because it is comfortable and it is reasonable for an option to be saved if selected.

Regards,
mantlabs
